frontend/packages/ceph-storage-plugin/src/components/ocs-install/node-list.tsx

```diff
@@ -0,0 +1,330 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import * as classNames from 'classnames';
+import { connect } from 'react-redux';
+import {
+  sortable,
+  SortByDirection,
+  ISortBy,
+  Table,
+  TableHeader,
+  TableBody,
+  TableVariant,
+} from '@patternfly/react-table';
+import { tableFilters } from '@console/internal/components/factory/table-filters';
+import { ActionGroup, Button } from '@patternfly/react-core';
+import { ButtonBar } from '@console/internal/components/utils/button-bar';
+import { history } from '@console/internal/components/utils/router';
+import {
+  SelectorInput,
+  humanizeCpuCores,
+  humanizeBinaryBytes,
+  ResourceLink,
+} from '@console/internal/components/utils/index';
+import {
+  getNodeRoles,
+  k8sCreate,
+  k8sPatch,
+  NodeKind,
+  referenceForModel,
+} from '@console/internal/module/k8s';
+import { NodeModel } from '@console/internal/models';
+import { Tooltip } from '@console/internal/components/utils/tooltip';
+import { OCSServiceModel } from '../../models';
+import { minSelectedNode, labelObj, ocsRequestData, taintObj } from '../../constants/ocs-install';
+import { OCSStorageClassDropdown } from './storage-class-dropdown';
+
+const ocsLabel = 'cluster.ocs.openshift.io/openshift-storage';
+
+const getConvertedUnits = (value, initialUnit) => {
+  return (
+    humanizeBinaryBytes(_.slice(value, 0, value.length - 2).join(''), initialUnit).string || '-'
+  );
+};
+
+const tableColumnClasses = [
+  classNames('col-md-1', 'col-sm-1', 'col-xs-2'),
+  classNames('col-md-5', 'col-sm-6', 'col-xs-8'),
+  classNames('col-md-2', 'col-sm-3', 'col-xs-2'),
+  classNames('col-md-2', 'hidden-sm', 'hidden-xs'),
+  classNames('col-md-2', 'col-sm-2', 'hidden-xs'),
+];
+
+const getColumns = () => {
+  return [
+    {
+      title: 'Name',
+      sortField: 'metadata.name',
+      transforms: [sortable],
+      props: { className: tableColumnClasses[1] },
+    },
+    {
+      title: 'Role',
+      props: { className: tableColumnClasses[2] },
+    },
+    {
+      title: 'CPU',
+      props: { className: tableColumnClasses[3] },
+    },
+    {
+      title: 'Memory',
+      props: { className: tableColumnClasses[4] },
+    },
+  ];
+};
+
+// return an empty array when there is no data
+const getRows = (nodes) => {
+  return nodes.map((node) => {
+    const roles = getNodeRoles(node).sort();
+    const obj = {
+      cells: [],
+      selected: false,
+      id: node.metadata.name,
+      metadata: _.clone(node.metadata),
+      spec: _.clone(node.spec),
+    };
+    obj.cells = [
+      {
+        title: <ResourceLink kind="Node" name={node.metadata.name} title={node.metadata.uid} />,
+      },
+      {
+        title: roles.join(', ') || '-',
+      },
+      {
+        title: `${humanizeCpuCores(_.get(node.status, 'capacity.cpu')).string || '-'} CPU`,
+      },
+      {
+        title: `${getConvertedUnits(_.get(node.status, 'allocatable.memory'), 'KiB')}`,
```

Comment:
Use the existing helpers in units.js instead of writing your own.

Reply:
@spadgett I am using `humanizeBinaryBytes` utils method inside `getConvertedUnits`. Are you referring to some other function?

Reply:
ack